specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/mfe.json

```diff
@@ -18041,54 +18871,235 @@
       "x-ms-discriminator-value": "Managed",
       "additionalProperties": false
     },
-    "ManagedIdentityCredential": {
-      "description": "Credential for user managed identity",
-      "type": "object",
+    "ManagedIdentityCredential": {
+      "description": "Credential for user managed identity",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataReferenceCredential"
+        }
+      ],
+      "properties": {
+        "managedIdentityType": {
+          "description": "ManagedIdentityCredential identity type",
+          "type": "string",
+          "x-nullable": true
+        },
+        "userManagedIdentityClientId": {
+          "description": "ClientId for the UAMI. For ManagedIdentityType = SystemManaged, this field is null.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "userManagedIdentityPrincipalId": {
+          "description": "PrincipalId for the UAMI. For ManagedIdentityType = SystemManaged, this field is null.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "userManagedIdentityResourceId": {
+          "description": "Full arm scope for the Id. For ManagedIdentityType = SystemManaged, this field is null.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "userManagedIdentityTenantId": {
+          "description": "TenantId for the UAMI. For ManagedIdentityType = SystemManaged, this field is null.",
+          "type": "string",
+          "x-nullable": true
+        }
+      },
+      "x-ms-discriminator-value": "ManagedIdentity",
+      "additionalProperties": false
+    },
+    "ManagedOnlineDeployment": {
+      "description": "Properties specific to a ManagedOnlineDeployment.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OnlineDeployment"
+        }
+      ],
+      "x-ms-discriminator-value": "Managed",
+      "additionalProperties": false
+    },
+    "MarketplacePlan": {
+      "additionalProperties": false,
+      "properties": {
+        "offerId": {
+          "description": "The Offer ID of the Marketplace Plan.",
+          "readOnly": true,
+          "type": "string",
+          "x-ms-mutability": [
+            "read"
+          ],
+          "x-nullable": true
+        },
+        "planId": {
+          "description": "The Plan ID of the Marketplace Plan.",
+          "readOnly": true,
+          "type": "string",
+          "x-ms-mutability": [
+            "read"
+          ],
+          "x-nullable": true
+        },
+        "publisherId": {
```

Comment:
>publisherId [](http://example.com/codeflow?start=9&length=11)

Please include info about the type/format of this ID in the description (name, GUID, resourceId, integer, etc).

Reply:
updated